File: app/plugins/ios-sources/SharedPrefsModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(SharedPrefs, NSObject)

RCT_EXTERN_METHOD(setAppGroupString:(NSString *)key value:(NSString *)value)
RCT_EXTERN_METHOD(setKeyboardData:(NSString *)encoded)
RCT_EXTERN_METHOD(setKeyboardTheme:(NSString *)theme)
RCT_EXTERN_METHOD(clearAppGroup)

@end

File: app/plugins/ios-sources/SharedPrefsModule.swift
import Foundation

@objc(SharedPrefs)
class SharedPrefsModule: NSObject {
  private let suiteName = "group.com.klavyem.shared"

  private var defaults: UserDefaults? {
    UserDefaults(suiteName: suiteName)
  }

  @objc func setAppGroupString(_ key: String, value: String) {
    defaults?.set(value, forKey: key)
    defaults?.synchronize()
  }

  @objc func setKeyboardData(_ encoded: String) {
    defaults?.set(encoded, forKey: "keyboard_data")
    defaults?.synchronize()
  }

  @objc func setKeyboardTheme(_ theme: String) {
    defaults?.set(theme, forKey: "keyboard_theme")
    defaults?.synchronize()
  }

  @objc func clearAppGroup() {
    guard let ud = defaults else { return }
    ud.dictionaryRepresentation().keys.forEach { ud.removeObject(forKey: $0) }
    ud.synchronize()
  }

  @objc static func requiresMainQueueSetup() -> Bool { false }
}
